owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt

```diff
@@ -1176,13 +1178,30 @@ class FileDisplayActivity : FileActivity(),
         when (uiResult) {
             is UIResult.Success -> {
                 when (uiResult.data) {
-                    SynchronizeFileUseCase.SyncType.AlreadySynchronized -> showSnackMessage(getString(R.string.sync_file_nothing_to_do_msg))
+                    SynchronizeFileUseCase.SyncType.AlreadySynchronized -> {
+                        if (waitingToPreview) {
+                            startPreview(fileWaitingToPreview)
+                            waitingToPreview = false
+                            fileWaitingToPreview = null
+                        } else {
+                            showSnackMessage(getString(R.string.sync_file_nothing_to_do_msg))
+                        }
+                    }
                     is SynchronizeFileUseCase.SyncType.ConflictDetected -> {
                         val showConflictActivityIntent = Intent(this, ConflictsResolveActivity::class.java)
                         showConflictActivityIntent.putExtra(ConflictsResolveActivity.EXTRA_FILE, file)
                         startActivity(showConflictActivityIntent)
                     }
-                    is SynchronizeFileUseCase.SyncType.DownloadEnqueued -> showSnackMessage("Download enqueued")
+                    is SynchronizeFileUseCase.SyncType.DownloadEnqueued -> {
+                        if (waitingToPreview) {
+                            showSnackMessage("A new version was found in server. Downloading...")
```

Comment:
Please, no hardcoded text. Make this translatable

Reply:
Thanks, didn't realize!